source/common/unified_malloc_framework/src/ipc.c

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "umf/ipc.h"
+
+#include "memory_pool_internal.h"
+#include "memory_tracker.h"
+
+#include <assert.h>
+#include <stdlib.h>
+
+struct umf_ipc_data_t {
+    uint32_t size;
+    uint64_t offset;
+    char providerData[];
+};
+
+enum umf_result_t
+umfGetIPCHandle(const void *ptr, umf_ipc_handle_t *umfIPCHandle, size_t *size) {
+    size_t ipcHandleSize = 0;
+    struct umf_alloc_info_t allocInfo;
+    enum umf_result_t res =
+        umfMemoryTrackerGetAllocInfo(umfMemoryTrackerGet(), ptr, &allocInfo);
+    if (res != UMF_RESULT_SUCCESS) {
+        return res;
+    }
+
+    // TODO: we have no cases with multiple memory providers
```

Comment:
ok but how would you implement case where we have more than single provider?

Reply:
we can extend memory tracker capabilities.

But the main question do we want to support multiple memory providers at all? That is why I keep this code with TODO comment. In the past when we discussed pool creation API with Igor, we briefly discussed whether it makes sense to allow multiple memory providers and what is the use case.

Reply:
what about the cases where the user wants a pool where some memory is allocated on CPU and the rest on disk? or GPU or sth else? Then you will have multiple providers.

Reply:
> what about the cases where the user wants a pool where some memory is allocated on CPU and the rest on disk? or GPU or sth else? Then you will have multiple providers.

I cannot imagine the use case for that. But even for such case, I think it should be still a single memory provider that is doing such magic internally. E.g. some special provider could be created that internally uses upstream memory providers for CPU, GPU, disk, etc. memory targets. The point is that from the pool perspective it will be a single pool manager and single memory provider.

Anyway, I think it is a topic for separate discussion, not for this PR. Let's talk about that in our sync.